source/tscript.js

```diff
@@ -7072,11 +7078,16 @@ module.Interpreter = function(program)
 	{
 		this.stop = true;
 	};
+	
+	// event limits
+	this.timerEventEnqueued = false;
 
 	// queue an event
+	// returns true when event got enqueued, false otherwise
 	this.enqueueEvent = function(type, event)
 	{
 		if (this.eventmode) this.eventqueue.push({"type": type, "event": event});
+		return this.eventmode;
 	};
```

Comment:
Little change to the `enqueueEvent` function, it now returns true if the event got enqueued, this currently only happens when `this.eventmode` is true.